STYLE_GUIDE.md

```diff
@@ -0,0 +1,190 @@
+    * Ideally, for faster merges, break your PR into several PRs that stack on top of each other and lead to `develop`.
+ * Reference branches
+   * Sometimes it is necessary to keep a branch with reference information even there there is no intention of merging
+   * These branches should be named `yourname/reference-ghi-description`, where `ghi` is the github issue number
+ * Pull Requests (PRs)
+   * Do not merge PRs that aren't final -- e.g. no code dead code or code that won't be used should be committed.
+   * Always delete branch immediately after merge
+   * When there are to many commits, [Squash](https://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git)
+   them to keep the history clean - don't do this where commits ought to be kept for future reference.
+   * All pull requests should include any relevant additions to the `CHANGELOG.md`, if the PR is opened against the
+     develop branch then include the changes under an `unreleased changes` header at the top of the document.
+
+## Linters
+
+These must be applied to all (go) repos.
+
+ * [shellcheck](https://github.com/koalaman/shellcheck)
+ * [gometalinter (covers all important linters)](https://github.com/alecthomas/gometalinter)
+   - See the `Makefile` in each repo for ongoing linting progress. Running `make metalinter` is the ultimate goal, while `make metalinter_test` enables all the linters that are currently passing, for integration with CI.
+   * gometalinter may make use of the [Tendermint fork of golint](https://github.com/tendermint/lint/)
+
+## Various
+
+ * Reserve "Save" and "Load" for long-running persistence operations. When parsing bytes, use "Encode" or "Decode".
+ * Receiver variables should be at least 2 letters.  e.g. not `func (s \*Something) MyFunc(){}` but `func (sm *Something) MyFunc(){}`.  This makes refactoring easier when we rename Something to something else.
+ * Maintain consistency across the codebase.
+ * Do not use "instance" in function names.
+ * Functions that return functions should have the suffix `Fn`
+ * A struct generally shouldn’t have a field named after itself, aka. this shouldn't occur:
+``` golang
+type middleware struct {
+	middleware Middleware
+}
+```
+ * In comments, use "iff" to mean, "if and only if".
+ * Product names are capitalized, like "Tendermint", "Basecoin", "Protobuf", etc except in command lines: `tendermint --help`
+ * Acronyms are all capitalized, like "RPC", "gRPC", "API".  "MyID", rather than "MyId".
+ * Prefer errors.New() instead of fmt.Errorf() unless you're actually using the format feature with arguments. (otherwise it's needlessly inefficient, & it'll crap out on %*)
+
+## Importing Libraries
+
+Sometimes it's necessary to rename libraries to avoid naming collisions or ambiguity.
+
+ * Use [goimports](https://godoc.org/golang.org/x/tools/cmd/goimports)
+ * Separate imports into blocks - one for the standard lib and one for others. In some cases, it's nice to
```

Comment:
I'd drop the "In some cases", and make this the general policy. Also, change "tendermint libs" to "application libs" or something - e.g. `tendermint/iavl` should be considered an external dependency in the `tendermint/tendermint` code base.